client/src/pages/Explore.js

```diff
@@ -0,0 +1,149 @@
+import moment from 'moment';
+
+import * as ExploreStyles from '../styles/ExploreSyles';
+import ProjectCard from '../components/ProjectCard';
+import Loading from '../components/Loading';
+import { withPageContext } from '../components/pageContext';
+import { OutlinedSelect } from '../components/Inputs';
+import { getUserProjects } from '../api/projects';
+import ProjectGrid from '../components/ProjectGrid';
+
+class Explore extends React.Component {
+  state = {
+    projects: [],
+    industries: [
+      'All Industries',
+      'Technology',
+      'Marketing',
+      'Engineering',
+      'Art',
+      'Film'
+    ],
+    locations: [
+      'Everywhere',
+      'New York',
+      'California',
+      'Georgia',
+      'Florida'
+    ],
+    industrySelect: 'All Industries',
+    locationSelect: 'Everywhere',
+    loading: true
+  };
+
+  async componentDidMount() {
+    const getProjects = await getUserProjects(this.props.userDetails.id);
+    const projects = getProjects.data;
+
+    this.setState({ projects }, () => {
+      setTimeout(() => this.setState({loading: false}), 500)
+    });
+  }
+
+  handleFilterSelects = (event) => {
+    const { target: { name, value } } = event;
+
+    this.setState({ [name]: value });
+  };
+
+  projectFilter = () => {
+    const { industrySelect, locationSelect } = this.state;
+
+    return this.state.projects.filter((project) => {
+      const industryCondition = project.industry === industrySelect;
+      const locationCondition = project.location.includes(locationSelect);
+
+      if (industrySelect !== 'All Industries' && locationSelect !== 'Everywhere') {
+        return industryCondition && locationCondition;
+      }
+
+      if (industrySelect !== 'All Industries') return industryCondition;
+
+      if (locationSelect !== 'Everywhere') return locationCondition;
+
+      return project;
+    });
+  };
+
+  renderGrid() {
+    const filteredPosts = this.projectFilter();
+
+    if (this.state.loading) return <Loading />;
+
+    return (
+      <ProjectGrid>
+          {filteredPosts.map(({_id: id, ...project}, index) => (
+            <ProjectCard
+              key={index}
+              onClick={() => this.props.history.push(`/projects/${id}`)}
+              creator={{
+                name: project.user.name,
+                location: project.location
+              }}
+              title={project.title}
+              image={project.images[0]}
+              funding={project.funding.fundingTotal}
+              fundingGoal={project.fundingGoal}
+              industry={project.industry}
+              daysLeft={parseInt(moment(project.fundingDeadline).fromNow().split(' ')[1])}
+            />
+          ))}
+
+        </ProjectGrid>
+    )
+  }
+
+  render() {
+    const { classes } = this.props;
+    const { industries, locations, industrySelect, locationSelect } = this.state;
+    const filteredPosts = this.projectFilter();
+
+    return (
+      <ExploreStyles.Main>
```

Comment:
Filters should conditionally render (not necessarily in the PR but at some point).

Reply:
I'll have to rethink maybe? by default, it is to show all

Reply:
see above edit. Changed "conditionally render" to "be responsive" was carrying that over from above. My apologies.